index.bs

```diff
@@ -279,5 +512,149 @@ Threat model {#security-threat}
 Mitigations and defenses {#security-defenses}
 ---------------------------------------------
 
+Appending A: Messages
+=====================
+
+Lower type keys are used for messages we expect to be more frequent
+and higher type keys are used for less frequently used messages.  This
+is because on the wire lower values encode smaller.
+
+<pre>
+; type key 14
+presentation-url-availability-request = {
+  request
+  1: [* text] ; urls
+  2: microseconds ; watch-duration
+  3: int ; watch-id
+}
+
+; type key 15
+presentation-url-availability-response = {
+  response
+  1: [* url-availability] ; url-availabilities
+}
+
+; type key 103
+presentation-url-availability-event = {
+  1: int ; watch-id
+  2: [* url] ; urls
+  3: [* url-availability] ; url-availabilities
+}
+
+
+; idea: use HTTP response codes?
+url-availability = &(
+  available: 0
+  unavailable: 1
+  invalid: 10
+)
+
+
+
+
+; type key 104
+presentation-start-request = {
+  request
+  1: text ; presentation-id
+  2: text ; url
+  3: text ; headers 
```

Comment:
[* text] or are these newline delimited?

Reply:
They are new-line delimited.  Basically, the straight bytes that should be written into the TCP/TLS socket.

Reply:
Of course, that doesn't really work for HTTP/3....  do we require impls to parse that or do we change to a list of (key, value)?

Reply:
OK, I changed to key/value pairs.